Sources/NIOHTTP2/ConnectionStateMachine/ConnectionStreamsState.swift

```diff
@@ -218,11 +220,28 @@ struct ConnectionStreamState {
         _ modifier: (inout HTTP2StreamStateMachine) -> StateMachineResultWithStreamEffect
     ) -> StateMachineResultWithStreamEffect {
         guard let result = self.activeStreams.autoClosingTransform(streamID: streamID, modifier) else {
-            // We never ignore recently reset streams here, as this should only ever be used when *sending* frames.
-            return StateMachineResultWithStreamEffect(
-                result: self.streamMissing(streamID: streamID, ignoreRecentlyReset: false, ignoreClosed: false),
-                effect: nil
-            )
+            // This state can be reached when a RST_STREAM frame is sent by this peer but the stream
+            // doesn't exist. This can happen for a few reasons, including:
+            //
+            // 1. The RST_STREAM frame is being sent because the stream wasn't accepted (i.e. the
+            //    client opening the stream raced with the server sending a GOAWAY frame). In this
+            //    case the state machine doesn't know about the stream but does need to send
+            //    a RST_STREAM frame.
+            // 2. An implementation or user error where the stream genuinely doesn't exist and
+            //    attempting to send a RST_STREAM frame is an error.
+            if streamID.isClientInitiated, streamID > self.lastClientStreamID {
```

Comment:
This check seems overly specific. Do we think it's not possible to get to this via PUSH_PROMISE?

Reply:
Oh, isn't this also asking the wrong question? This would also satisfy if _we_ were the client, which seems wrong.